### PR TITLE
Refactor spec use expect_correction in Lint cops

### PR DESCRIPTION
**Pull request by biinari**
Part of #8127

Use newer expect_correction API in Lint namespace cops.

Move expected corrections to directly follow expected offenses.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [ ] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by marcandre** [comment]
I think we should tweak `expect_offense` so that we can abbreviate the error message, say with by putting `[...]` at the end of it.

File: spec/rubocop/cop/lint/empty_when_spec.rb [hunk 1]
```diff
@@ -1,55 +1,37 @@
...
+      expect_no_offenses(code)
     end
   end
 
   let(:cop_config) { { 'AllowComments' => false } }
 
-  let(:message) { 'Avoid `when` branches without a body.' }
-
   context 'when a `when` body is missing' do
     it_behaves_like 'code with offense', <<~RUBY
```
[the remark this answers is not in the document]
Comment (biinari) [reply spec/rubocop/cop/lint/empty_when_spec.rb:22]: No problem, first pass was just trying to reduce the change noise but I agree the `shared_examples` not needed here.

**Comment by biinari** [comment]
I'm going to go with matching the `message = ...` pattern used in the rest of the files touched here with long offense messages (for consistency within the file and hopefully for improved readability).

**Comment by marcandre** [comment]
💪 Lots of great work in there, thanks! 🎉

Result looks super clean. Great job.

BTW, we can now abbreviate the error messages with `[...]`; no need to amend this PR but if you continue cleaning up other specs...

**Comment by biinari** [comment]
Thanks @marcandre. I'm going to take a look at the style namespace cops this weekend (probably split into ranges to keep PR size reasonable).

I'll make use of the `[...]` abbreviation in those. Nice work getting that in.

**Pull request merged**